feat(get movies): add get endpoint to retrieve movies with filters

sachinmurali opened the pull request:
**What:** 

Stands up an endpoint to retrieve a list of movies based on a few filters.

**Why:** 

This is a core component of the sfmovies onboarding project - https://lobsters.atlassian.net/browse/DE-262

**Details:**

This PR sets up the `/GET` movies endpoint to retrieve a list of movies from the DB. The endpoint accepts 5 parameters - 

- `title`: movie title as a string (required and always accompanied by `fuzzy_title`),
- `fuzzy_title`: a boolean indicating whether or not we need to perform a fuzzy match,
- `release_year`: an integer indicating the year the said movie is released,
- `after`: an integer specifying the start year,
- `before`: an integer specifying the end year,

`after` and `before` indicate the range of years.

**Notes:**

To test this change, pull down the branch and run `yarn test`. Additionally, you can create a movie with the post endpoint from https://github.com/sachinmurali/sfmovies/pull/5 and try to retrieve the newly created movie.

If I were to create a movie called `Inception`, you can stand up the server via `yarn start` and run a `url --location --request GET 'http://localhost:3000/movies?fuzzy_title=True&title=ion'`. You can provide different combination of filters (mentioned above) to acheive the desired result.

Commit:
feat(get movies): add get endpoint to retrieve movies with filters

javillanueva1 commented on test/plugins/features/movies/controller.test.js:
```diff
@@ -0,0 +1,126 @@
+
+  });
+
+  describe('get all movies', () => {
```
I'd say that you should group all of these get functions rather than have a describe block for each way that the GET function is used

sachinmurali replied:
That makes sense! 👍

javillanueva1 commented on test/plugins/features/movies/controller.test.js:
```diff
@@ -0,0 +1,126 @@
+
+  });
+
+  describe('create', () => {
```
Rather than `create` here, I'd use `POST` as you're describing the POST endpoint

sachinmurali replied:
Isn't this at the controller level though? I'm using `POST` at the `index` level.

javillanueva1 replied:
Ahh, yeah, I'm sorry, that makes sense

javillanueva1 reviewed (commented):
Looking good, there are a few comments below

javillanueva1 commented:
I realize that for this and your other outstanding PR) but you should follow the PR title format guidelines here in the case that you don't have a Jira to follow this with (this should be a rare situation, but it's good to know how to do this anyways): https://www.notion.so/lob/Code-Review-Guidelines-aad42d64cb724f3cadd1ab233bfc93df#ce139b836f7442d7b835318e4afae3a4

sachinmurali commented:
@javillanueva1 Thanks for pointing that out :) I've updated the description per the notion doc and rebased off of master after merging the post endpoint.

javillanueva1 reviewed (approved):
LGTM
